src/cli/self_update.rs

```diff
@@ -197,46 +148,135 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         }
     }
 
-    eprintln!(
-        "{}Pixi archive downloaded.",
-        console::style(console::Emoji("✔ ", "")).green(),
-    );
+    Ok(archived_tempfile)
+}
 
+/// Unpacks a pixi archive (typically downloaded from GitHub releases) into a temporary directory.
+async fn unpack_release_archive(
+    mut archived_tempfile: NamedTempFile,
```

Comment:
same here - we can just accept something that implements AsyncRead and use it for unpacking